kernel/sched.c

```diff
@@ -1242,6 +1242,9 @@ static inline void z_vrfy_k_wakeup(k_tid_t thread)
 
 k_tid_t z_impl_k_sched_current_thread_query(void)
 {
+#ifdef CONFIG_ARCH_HAS_CUSTOM_CURRENT_IMPL
+	return arch_current_thread();
+#else
 #ifdef CONFIG_SMP
```

Comment:
probably can just call `arch_current_thread()` here, and use the current implementation as the default when `CONFIG_ARCH_HAS_CUSTOM_CURRENT_IMPL=n`